src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/CurrentAppDomainExceptionProcessor.cs

```diff
@@ -48,5 +49,16 @@ private void ConfigureUnhandledPipeline(ExceptionPipelineBuilder builder)
             // Report exception through event source
             builder.Add(next => new UnhandledExceptionEventsPipelineStep(next, _eventSource, _idSource).Invoke);
         }
+
+        public void Dispose()
+        {
+            _eventSource.Dispose();
```

Comment:
The event source should be disposed last since the pipelines use the event source.